### PR TITLE
V3/controllers harvest auth

### PR DESCRIPTION
**Pull request by leefrank9527**
Fixed Spring Hibernate issues related to Harvest Authorising.

### CODE REVIEW

File: src/main/java/org/webcurator/domain/UserRoleDAOImpl.java [hunk 1]
```diff
@@ -122,7 +122,12 @@ public List getUsers(Long agencyOid) {
 
 
     public List getAgencies() {
+//<<<<<<< Updated upstream
         Query q = sessionFactory.getCurrentSession().createNamedQuery(Agency.QRY_GET_ALL_AGENCIES);
+//=======
+////        Query q = sessionFactory.getCurrentSession().createQuery(Agency.QRY_GET_ALL_AGENCIES);
+//        Query q = sessionFactory.getCurrentSession().createNamedQuery(Agency.QRY_GET_ALL_AGENCIES,Agency.class);
+//>>>>>>> Stashed changes
```
[the remark this answers is not in the document]
Comment (leefrank9527) [reply src/main/java/org/webcurator/domain/UserRoleDAOImpl.java]: Accepted. Cleaned the outlined redundant comments.

File: src/main/java/org/webcurator/domain/model/core/UrlPattern.java [hunk 2]
```diff
@@ -23,150 +23,167 @@
...
+    @JoinColumn(name = "UP_SITE_ID", foreignKey = @ForeignKey(name = "FK_UP_SITE_ID"))
+    private Site site;
+    /**
+     * A set of permissions.
+     */
+    //CascadeType.PERSIST, CascadeType.REFRESH, CascadeType.MERGE
+//    @JoinTable(name = "PERMISSION_URLPATTERN",
+//            joinColumns = {@JoinColumn(name = "PU_URLPATTERN_ID")},
+//            inverseJoinColumns = {@JoinColumn(name = "PU_PERMISSION_ID")},
+//            foreignKey = @ForeignKey(name = "PU_FK_2"))
```
[the remark this answers is not in the document]
Comment (leefrank9527) [reply src/main/java/org/webcurator/domain/model/core/UrlPattern.java]: Accepted. Cleaned the outlined redundant comments.

**Comment by obrienben** [comment]
@leefrank9527 anything else to include in this PR or is it done?

**Comment by frank-dia** [comment]
It's done.

**Pull request merged**